### ISSUE TITLE
run: cmdline args processing forbids passing --version to the cmd

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#3067]
```shell
$> datalad run heudiconv --version 
datalad 0.11.1.dev55

$> datalad run --cmd heudiconv --version 
datalad 0.11.1.dev55

$> datalad run -- heudiconv --version   
[INFO   ] == Command start (output follows) ===== 
/bin/sh: 0: Illegal option --
[INFO   ] == Command exit (modification check follows) ===== 
[INFO   ] The command had a non-zero exit code. If this is expected, you can save the changes with 'datalad save -r -F .git/COMMIT_EDITMSG .' 
CommandError: command '-- heudiconv --version' failed with exitcode 2
Failed to run u'-- heudiconv --version' under '/tmp/testds'. Exit code=2.
```
I guess it is the side-effect of some of my "optimizations" to make cmdline invocations of basic commands tolerably not-slow.  Seems to be --version specific since doesn't happen for `--help`.

Also -- may be it is worthwhile to add handling of `--` as an indicator for "do not consider any subsequent argument as a possible option and treat as a positional argument" which is I believe a common convention (don't remember where it stems from)

**Comment by kyleam** [comment]
> Seems to be --version specific since doesn't happen for `--help`.

Odd.

> Also -- may be it is worthwhile to add handling of `--` as an indicator for "do not consider any subsequent argument as a possible option and treat as a positional argument" which is I believe a common convention (don't remember where it stems from)

I think it'd be worthwhile because it would address datalad/datalad-container#46.  I don't particularly like promoting it as a solution to this issue, which I think this really should be viewed as an issue with datalad's parser and fixed there.

To add a working variant to your list:

    $ datalad run 'heudiconv --version'

**Comment by yarikoptic** [comment]
FWIW - argparse seems to do handling of `--` internally (at least in `_parse_known_args`):
```
$> datalad run blah --version   
datalad 0.11.1.dev60

$> datalad run git --version    
datalad 0.11.1.dev60

$> datalad run git -- --version 
[INFO   ] == Command start (output follows) ===== 
unknown option: --
usage: git [--version] [--help] [-C <path>] [-c <name>=<value>]
           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
           [-p | --paginate | -P | --no-pager] [--no-replace-objects] [--bare]
           [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
           <command> [<args>]
[INFO   ] == Command exit (modification check follows) ===== 
[INFO   ] The command had a non-zero exit code. If this is expected, you can save the changes with 'datalad save -r -F .git/COMMIT_EDITMSG .' 
CommandError: command 'git -- --version' failed with exitcode 129
Failed to run u'git -- --version' under '/tmp/testds'. Exit code=129.
```
but then doesn't seems to strip it from the list of the arguments. [comment on SO](https://stackoverflow.com/questions/9084080/leave-arguments-untouched-with-argparse#comment93430264_9084137) suggests adding `--` as an explicit argument with a proper destination to swallow it, which sounds exactly what `--cmd` is there for (without any special handling from argparser)... tried a few things but didn't succeed in making it sensible yet

**Comment by kyleam** [comment]
I said:
> I don't particularly like promoting it as a solution to this issue, which I think should be viewed as an issue with datalad's parser and fixed there.

I still agree with this, but this issue was opened as a `run` issue and that part is addressed with 7f9fe2e3c (ENH: run: Strip leading "--" to allow command disambiguation, 2019-01-11).  I don't think there is much value in keeping this open.
